src/main/java/uk/gov/hmcts/reform/divorce/orchestration/tasks/PopulateDocLink.java

```diff
@@ -10,32 +10,33 @@
 import java.util.Map;
 
 import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.DOCUMENT_TYPE;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.SOL_DOCUMENT_LINK_FIELD;
 
 @Component
-public class PopulateMiniPetitionUrl implements Task<Map<String, Object>>  {
+public class PopulateDocLink implements Task<Map<String, Object>>  {
 
     private static final String D8DOCUMENTS_GENERATED = "D8DocumentsGenerated";
     private static final String VALUE = "value";
-    private static final String DOCUMENT_TYPE = "DocumentType";
-    private static final String PETITION_TYPE = "petition";
-
-    private static final String DOCUMENT_LINK = "DocumentLink";
-    private static final String MINI_PETITION_LINK = "minipetitionlink";
+    private static final String DOCUMENT_TYPE_KEY = "DocumentType";
+    private static final String DOCUMENT_LINK_KEY = "DocumentLink";
 
     @SuppressWarnings("unchecked")
     @Override
     public Map<String, Object> execute(TaskContext context, Map<String, Object> payload) throws TaskException {
         List<Map> documentList =
                 ofNullable(payload.get(D8DOCUMENTS_GENERATED)).map(i -> (List<Map>) i).orElse(new ArrayList<>());
 
+        String documentType = context.getTransientObject(DOCUMENT_TYPE);
+        String docLinkFieldName = context.getTransientObject(SOL_DOCUMENT_LINK_FIELD);
         Map<String, Object> petitionDocument = documentList
                 .stream()
-                .filter(map -> PETITION_TYPE.equals(((Map) map.get(VALUE)).get(DOCUMENT_TYPE)))
+                .filter(map -> documentType.equals(((Map) map.get(VALUE)).get(DOCUMENT_TYPE_KEY)))
```

Comment:
I think you could have a map operation (or two) for ((Map) map.get(VALUE) and another for the .get(DOCUMENT_TYPE_KEY).
```suggestion
                .map(map -> map.get(VALUE))
                .map(Map.class::cast)
                .map(map -> map.get(DOCUMENT_TYPE_KEY))
                .filter(documentType::equals)
```

Not sure it works...